### PR TITLE
BUG: fix for some RuntimeWarnings with BetaGeoFitter

### PR DESCRIPTION
**Pull request by AdriMarteau**
When I was using the `BetaGeoFitter` class on some data I ran into a couple of `RuntimeWarning`s due to negative logarithms or overflowed exponentials (like #25) especially with very high value for `frequency`. 

I have modified 3 functions to avoid those warnings:
- `_negative_log_likelihood`
- `conditional_expected_number_of_purchases_up_to_time`
- `conditional_probability_alive` 

I have also included a test for testing that no overflow happens in `conditional_probability_alive`

### CODE REVIEW

File: tests/test_estimation.py [hunk 1]
```diff
@@ -513,6 +513,20 @@ def test_fit_with_index(self, cdnow_customers):
         )
         assert (bgf.data.index == index).all() == False
 
+    def test_no_runtime_warnings_high_frequency(self, cdnow_customers):
```
Comment (CamDavidsonPilon) [inline tests/test_estimation.py]: 👍 nice test

**Comment by CamDavidsonPilon** [comment]
This is great, thanks for this. Merging now

File: lifetimes/estimation.py [hunk 2]
```diff
@@ -699,8 +708,16 @@ def conditional_probability_alive(self, frequency, recency, T):
         Returns: a scalar
 
         """
+        ln_exp_max = 300.0
+
         r, alpha, a, b = self._unload_params('r', 'alpha', 'a', 'b')
-        return 1. / (1 + (frequency > 0) * (a / (b + frequency - 1)) * ((alpha + T) / (alpha + recency)) ** (r + frequency))
+        log_div = (r + frequency) * log(
```
Comment (statwonk) [inline lifetimes/estimation.py]: Nice fix @AdriMarteau! I was reading through the associated paper and wondering why the `log` were necessary. :-D
